src/main/scala/tech/cryptonomic/conseil/tezos/michelson/dto/MichelsonInstruction.scala

```diff
@@ -32,8 +34,17 @@ case class MichelsonSingleInstruction(
 /* Class representing a sequence of Michelson instructions */
 case class MichelsonInstructionSequence(instructions: List[MichelsonInstruction] = List.empty)
```

Comment:
Might not be the best idea to spend time on this... but, since the parser will never call this with an empty list arg, why not changing the constructor to only accept non empty lists? varargs can be used to do that.
Would this avoid the need for a normalized form wholesale?